src/main/java/org/dsmhack/config/JWTAuthenticationFilter.java

```diff
@@ -58,11 +60,16 @@ protected void successfulAuthentication(HttpServletRequest request,
                                             FilterChain chain,
                                             Authentication auth) throws IOException, ServletException {
 
+        String userGuid = ((User) auth.getPrincipal()).getUsername();
         String token = Jwts.builder()
-                .setSubject(((User) auth.getPrincipal()).getUsername())
+                .setSubject(userGuid)
                 .setExpiration(new Date(System.currentTimeMillis() + THIRTY_MINUTES))
                 .signWith(SignatureAlgorithm.HS512, jwtEncryptionKey.getBytes())
                 .compact();
         response.addHeader(HEADER_STRING, TOKEN_PREFIX + token);
+
+        response.getWriter().write(userRepository.findOne(userGuid).toJson());
+        response.getWriter().flush();
+        response.getWriter().close();
```

Comment:
I'm not very familiar with `HttpServletResponse`, I found how to write to the body here:
https://stackoverflow.com/questions/7658881/setting-a-string-in-a-body-of-httpresponse